draft-ietf-quic-tls.md

```diff
@@ -255,142 +205,197 @@ A simplified TLS 1.3 handshake with 0-RTT application data is shown in
+
+Rather than a strict layering, these two protocols are co-dependent: QUIC uses
+the TLS handshake; TLS uses the reliability and ordered delivery provided by
+QUIC streams.
+
+At a high level, there are two main interactions between the TLS and QUIC
+components:
+
+* The TLS component sends and receives messages via the QUIC component, with
+  QUIC providing a reliable stream abstraction to TLS.
+
+* The TLS component provides a series of updates to the QUIC
+  component, including (a) new packet protection keys to install (b)
+  state changes such as handshake completion, the server certificate,
+  etc.
+
+{{schematic}} shows these interactions in more detail, with the QUIC
+packet protection being called out specially.
+
+~~~
++------------+                        +------------+
+|            |<- Handshake Messages ->|            |
+|            |<---- 0-RTT Keys -------|            |
+|            |<--- Handshake Keys-----|            |
+|   QUIC     |<---- 1-RTT Keys -------|    TLS     |
+|            |<--- Handshake Done ----|            |
++------------+                        +------------+
+ |         ^
+ | Protect | Protected
+ v         | Packet
++------------+
+|   QUIC     |
+|  Packet    |
+| Protection |
++------------+
+~~~
+{: #schematic title="QUIC and TLS Interactions"}
+
 
- * A pre-shared key mode can be used for subsequent handshakes to reduce the
-   number of public key operations.  This is the basis for 0-RTT data, even if
-   the remainder of the connection is protected by a new Diffie-Hellman
-   exchange.
 
+# Carrying TLS Messages {#carrying-tls}
 
-# TLS Usage
+QUIC carries TLS handshake data in CRYPTO frames, each of which
+consists of a contiguous block of handshake data (identified by an
+offset and length). Those frames are packaged into QUIC packets
+and encrypted under the current TLS encryption level.
+As with TLS over TCP, once TLS handshake data has
+been delivered to QUIC, it is QUIC's responsibility to deliver it
+reliably. Each chunk of data is associated with the then-current TLS
+sending keys, and if QUIC needs to retransmit that data, it MUST use
+the same keys even if TLS has already updated to newer keys.
 
-QUIC reserves stream 0 for a TLS connection.  Stream 0 contains a complete TLS
-connection, which includes the TLS record layer.  Other than the definition of a
-QUIC-specific extension (see {{quic_parameters}}), TLS is unmodified for this
-use.  This means that TLS will apply confidentiality and integrity protection to
-its records.  In particular, TLS record protection is what provides
-confidentiality protection for the TLS handshake messages sent by the server.
+One important difference between TLS 1.3 records (used with TCP)
+and QUIC CRYPTO frames is that in QUIC multiple frames may appear
+in the same QUIC packet as long as they are associated with the
+same encryption level. For instance, an implementation might
+bundle a Handshake message and an ACK for some Handshake
```

Comment:
suggestion: "For instance, an implementation may bundle a CRYPTO frame and an ACK for some handshake packets into the same packet."